runtime/reconcilers/alert.go

```diff
@@ -882,12 +887,34 @@ func calculateExecutionTimes(a *runtimev1.Alert, watermark, previousWatermark ti
 
 // extractQueryResultFirstRow extracts the first row from a query result.
 // TODO: This should function more like an export, i.e. use dimension/measure labels instead of names.
-func extractQueryResultFirstRow(q runtime.Query) (map[string]any, bool, error) {
+func extractQueryResultFirstRow(q runtime.Query, measures []*runtimev1.MetricsViewSpec_MeasureV2, logger *zap.Logger) (map[string]any, bool, error) {
 	switch q := q.(type) {
 	case *queries.MetricsViewAggregation:
 		if q.Result != nil && len(q.Result.Data) > 0 {
 			row := q.Result.Data[0]
-			return row.AsMap(), true, nil
+			m := row.AsMap()
+			res := make(map[string]any)
+			for k, v := range m {
+				var measureLabel string
+				var f formatter.Formatter
+				for _, measure := range measures {
+					if k == measure.Name {
+						measureLabel = measure.Label
+						var err error
+						// D3 formatting isn't implemented yet so using the format preset for now
+						f, err = formatter.NewPresetFormatter(measure.FormatPreset, false)
+						if err != nil {
+							return nil, false, err
+						}
+						break
+					}
+				}
+				if measureLabel == "" {
+					measureLabel = k
+				}
+				res[measureLabel] = formatValue(f, v, logger)
+			}
+			return res, true, nil
```

Comment:
1. Can we pull this out into a util function to avoid too much nesting? E.g. `func formatResultRow(row map[string]any, metricsView) map[string]any`
2. Do you think it would make sense to log the error from `NewPresetFormatter` and output the non-formatted values? Instead of failing the alert due to an error in the format preset.